constraint/pkg/client/drivers/local/local.go

```diff
@@ -186,7 +197,7 @@ func (d *driver) alterModules(ctx context.Context, insert insertParam, remove []
 		}
 	}
 
-	c := ast.NewCompiler().WithPathConflictsCheck(storage.NonEmpty(ctx, d.storage, txn))
+	c := ast.NewCompiler().WithPathConflictsCheck(storage.NonEmpty(ctx, d.storage, txn)).WithUnsafeBuiltins(d.unsafeBuiltin)
```

Comment:
We should consider whether users want to explicitly specify which builtins they wish to allow, or if they want to disallow builtins.

The former is more secure, the latter requires more management, so I suspect we'll hear from people wanting both.

https://github.com/open-policy-agent/opa/blob/43a12ecea4eea9256695ac91d95b8bb7af3e6e72/ast/capabilities.go#L29-L46

Seems like the place to look to get a set of all builtins if we want to enable a pattern of selective disabling.

Reply:
> Looks like `WithUnsafeBuiltins` is deprecated, can we use `WithCapabilities`? https://pkg.go.dev/github.com/open-policy-agent/opa/ast#Compiler.WithUnsafeBuiltins

Make a change with `WithCapabilities` instead, but I find it's less preferable than `WithUnsafeBuiltins` in this usage:
-  the error message `undefined function http.send` is not as specific as [unsafe built-in function](https://github.com/open-policy-agent/opa/blob/master/ast/compile.go#L3933)
-  the usage of setting Capabilities could potentially overwrite (or be overwritten by) user-supplied Capabilities

Reply:
Can you expand on the overwriting comment?

Because the Rego compiler is private to the driver, I'm not sure how a user would be able to bypass our logic?

Reply:
Maybe we need to export the [capabilities](https://github.com/open-policy-agent/opa/blob/v0.27.1/ast/compile.go#L102) field in compiler. 
Say in the future the driver needs to both set custom Builtins and disable unsafe Builtins through `WithCapabilities`, they're only able to provide input base on `CapabilitiesForThisVersion` or `LoadCapabilitiesJSON`, the earlier set would be overwritten.

Reply:
If that becomes necessary in the future, could we export a functional argument that lets people define capabilities, like this:

https://github.com/open-policy-agent/gatekeeper/blob/6d83b8c173b34ac88afe18d7cf2808bf178789b7/main.go#L234

But extended with this:

`driver := local.New(local.Tracing(false), local.HTTPBuiltinEnabled(false), local.ExtraBuiltins([some list of builtins]))`

?

Reply:
Going along with Max's suggestion, I added a `capabilities` field in driver to manage its own builtin configs.